pyscf/adc/radc.py

```diff
@@ -529,12 +554,20 @@ def kernel_gs(self):
         mem_incore = (max(nao_pair**2, nmo**4) + nmo_pair**2) * 8/1e6
         mem_now = lib.current_memory()[0]
 
-        if (self._scf._eri is None or
-            (mem_incore+mem_now >= self.max_memory and not self.incore_complete)):
+        if getattr(self, 'with_df', None) or getattr(self._scf, 'with_df', None):  
+           if getattr(self, 'with_df', None): 
+               self.with_df = self.with_df
+           else :
+               self.with_df = self._scf.with_df
```

Comment:
Basically, I am confused by the use of self.with_df. I think the logic should be: if SCF is DF, then ADC must be DF. If SCF is exact, then ADC can be either DF or exact. In this case self.with_df should be False by default and should be switched to True if SCF is DF or SCF is exact and user wants ADC to be DF. Thus, we shouldn't need if-else statement here, because the "if getattr(self, 'with_df', None) or getattr(self._scf, 'with_df', None)" already implies that ADC must be DF and we should just set self.with_df to True.